src/tests/index.test.js

```diff
@@ -1,16 +1,476 @@
 'use strict';
 
 var expect = require('expect.js'),
-    Plugin = require('../index.js');
+    Plugin = require('../index.js'),
+    BbPromise = require('bluebird'),
+    sinon = require('sinon');
 
 describe('serverless-plugin-external-sns-events', function() {
 
+   function createMockServerless(requestFunc) {
+
+      var serverless, provider;
+
+      provider = {
+         request: requestFunc
+      };
+
+      serverless = {
+         getProvider: function(providerName) {
+
+            if (providerName !== 'aws') {
+               return null;
+            }
+
+            return provider;
+         },
+         service: {
+            provider: {
+               compiledCloudFormationTemplate: {
+                  Resources: {}
+               }
+            }
+         },
+         cli: { log: function() {
+            return;
+         } }
+         // cli: { log: function(val) {
+         //    process.stdout.write(val + '\n');
+         // } }
+      };
+
+      return serverless;
+
+   }
+
+   function createMockRequest(requestStub) {
+
+      return function() {
+
+         var reqArgs = Array.prototype.slice.call(arguments);
+
+
+         return new BbPromise(function(resolve, reject) {
+            var result;
+
+            result = requestStub.apply(undefined, reqArgs);
+            if (result !== null) {
+               resolve(result);
+               return;
+            }
+
+            reject(new Error('Call to request() with unexpected arguments:  ' +
+            JSON.stringify(reqArgs)));
+
+         });
+
+      };
+
+   }
+
+   function isPromise(obj) {
+      return !!obj && (typeof obj === 'object' || typeof obj === 'function') && typeof obj.then === 'function';
+   }
+
    describe('addEventPermission', function() {
 
-      // TODO: write tests
+      it('can compile lambda permission with correct FunctionName and SourceArn', function() {
+
+         // ARRANGE:
```

Comment:
The "ARRANGE", "ACT", "ASSERT" comments don't really add much. The purpose of a comment should be to explain what's happening, but this needs no explanation - at least not a simple one that's merely defining the three stages inherent to every unit test.